numba/cuda/simulator/vector_types.py

```diff
@@ -0,0 +1,60 @@
+from numba import types
+from numba.cuda.stubs import _vector_type_stubs
+
+
+class SimulatedVectorType:
+    attributes = ['x', 'y', 'z', 'w']
+
+    def __init__(self, *args):
+        args_flattened = []
+        for arg in args:
+            if isinstance(arg, SimulatedVectorType):
+                args_flattened += arg.as_list()
+            else:
+                args_flattened.append(arg)
+        self._attrs = self.attributes[:len(args_flattened)]
+        if not self.num_elements == len(args_flattened):
+            raise TypeError(
+                f"{self.name} expects {self.num_elements}"
+                f" elements, got {len(args_flattened)}"
+            )
+        assert self.num_elements == len(args_flattened), \
+            f"{self.num_elements} != {len(args_flattened)}"
```

Comment:
I think this is now superfluous given the check and raise above?

Reply:
Oops yes.